src/node_env_var.cc

```diff
@@ -385,8 +386,43 @@ static void EnvDeleter(Local<Name> property,
 static void EnvEnumerator(const PropertyCallbackInfo<Array>& info) {
   Environment* env = Environment::GetCurrent(info);
 
-  info.GetReturnValue().Set(
-      env->env_vars()->Enumerate(env->isolate()));
+  info.GetReturnValue().Set(env->env_vars()->Enumerate(env->isolate()));
+}
+
+static void EnvDefiner(Local<Name> property,
+                       const PropertyDescriptor& desc,
+                       const PropertyCallbackInfo<Value>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  if (desc.has_get() || desc.has_set()) {
+    env->ThrowError("Cannot use getter or setter on 'process.env'");
+  } else if (desc.has_value()) {
+    // calling env->EmitProcessEnvWarning() sets a variable indicating that
```

Comment:
Do you think you could call `EnvSetter(property, desc.value(), info);` in this case instead, to avoid duplicating code?